src/code/sys_math3d.c

```diff
@@ -0,0 +1,1691 @@
+    temp_f12 = vec2->y * dotProduct;
+    temp_f14 = vec2->z * dotProduct;
+
+    ret->x = ((temp_f2 + vec1->x) + (temp_f2 + vec1->x)) + negVec1.x;
+    ret->y = ((temp_f12 + vec1->y) + (temp_f12 + vec1->y)) + negVec1.y;
+    ret->z = ((temp_f14 + vec1->z) + (temp_f14 + vec1->z)) + negVec1.z;
+}
+
+s32 func_800CB198(f32 arg0, f32 arg1, f32 arg2, f32 arg3, f32 arg4, f32 arg5) {
+    if (arg0 <= arg4 && arg4 <= arg1 && arg2 <= arg5 && arg5 <= arg3) {
+        return 1;
+    }
+    return 0;
+}
+
+/*************************************************************************/
+/*         The next 2 functions have some interesting control flow       */
+/*************************************************************************/
+#pragma GLOBAL_ASM("asm/non_matchings/code/sys_math3d/func_800CB1F8.s")
+
+#ifdef NON_MATCHING
+/*
+ * Math3D_TriInSphere
+ * Checks if a tringle defined by `v0`, `v1`, and `v2` lies within a spehere
+ * centered at `center` and has radius `radius`.  Returns 1 if any vertex of the
+ * triangle lies within the sphere, or 0 otherwise.
+ */
+s32 func_800CB338(Vec3f* v0, Vec3f* v1, Vec3f* v2, Vec3f* center, f32 radius) {
+    f32 minX;
+    f32 maxX;
+    f32 minY;
+    f32 maxY;
+    f32 minZ;
+    f32 maxZ;
+
+    if (v1->x < v0->x) {
+        minX = v1->x;
+        maxX = v0->x;
+    } else if (v0->x < v1->x) {
+        minX = v0->x;
+        maxX = v1->x;
+    }
+
+    if (v1->y < v0->y) {
+        minY = v1->y;
+        maxY = v0->y;
+    } else if (v0->y < v1->y) {
+        minY = v0->y;
+        maxY = v1->y;
+    }
+
+    if (v1->z < v0->z) {
+        minZ = v1->z;
+        maxZ = v0->z;
+    } else if (v0->z < v1->z) {
+        minZ = v0->z;
+        maxZ = v1->z;
+    }
+
+    if (v2->x < minX) {
+        minX = v2->x;
+    } else if (maxX < v2->x) {
+        maxX = v2->x;
+    }
+
+    if (v2->y < minY) {
+        minY = v2->y;
+    } else if (maxY < v2->y) {
+        maxY = v2->y;
+    }
+
+    if (v2->z < minZ) {
+        minZ = v2->z;
+    } else if (maxZ < v2->z) {
+        maxZ = v2->z;
+    }
+
+    if (((minX - radius) <= center->x) && (center->x <= (maxX + radius)) && ((minY - radius) <= center->y) &&
+        (center->y <= (maxY + radius)) && ((minZ - radius) <= center->z) && (center->z <= (maxZ + radius))) {
+        return 1;
+    }
+
+    return 0;
+}
+#else
+#pragma GLOBAL_ASM("asm/non_matchings/code/sys_math3d/func_800CB338.s")
+#endif
+/**********************************************************************/
+
+f32 func_800CB55C(f32 arg0, f32 arg1) {
+    return SQ(arg0) + SQ(arg1);
+}
+
+f32 func_800CB570(f32 arg0, f32 arg1) {
+    return sqrtf(func_800CB55C(arg0, arg1));
+}
+
+f32 func_800CB594(f32 arg0, f32 arg1, f32 arg2, f32 arg3) {
+    func_800CB55C(arg0 - arg2, arg1 - arg3);
+}
+
+f32 func_800CB5D8(f32 arg0, f32 arg1, f32 arg2, f32 arg3) {
+    return sqrtf(func_800CB594(arg0, arg1, arg2, arg3));
+}
+
+f32 Math3D_Vec3f_HadamardProduct(Vec3f* vec) {
+    return SQ(vec->x) + SQ(vec->y) + SQ(vec->z);
+}
```

Comment:
This isn't a Hadamard Product, it's a dot product of vec with itself.
(A Hadamard product operates on two matrices of the same dimensions and produces a new matrix of the same dimensions by element-wise multiplication, but this function produces a scalar out of a vector through a dot product with itself.)